tests/integration/test_david_map_reduce_weighted.py

```diff
@@ -53,11 +53,11 @@ def test_david_map_reduce_weighted(self) -> None:
             )
             out_path = Path(tmp_dir) / "david_coreset.png"
             mmd_coreset, mmd_random = david_map_reduce_weighted_main(
-                in_path=in_path, out_path=out_path
+                in_path=in_path, out_path=out_path, downsampling_factor=4
             )
 
             self.assertEqual(
-                call("Image dimensions: (215, 180)"),
+                call("Image dimensions: (53, 45)"),
```

Comment:
Very minor - but could you just add a comment explaining that we expect (53, 45) because they are the original image size (215, 180) divided by the downsampling factor of 4 (then rounded)?